src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -14,21 +14,37 @@
 
 namespace Aspire.Dashboard;
 
-public class DashboardWebApplication
+public class DashboardWebApplication : IAsyncDisposable
 {
     private const string DashboardOtlpUrlVariableName = "DOTNET_DASHBOARD_OTLP_ENDPOINT_URL";
     private const string DashboardOtlpUrlDefaultValue = "http://localhost:18889";
     private const string DashboardUrlVariableName = "ASPNETCORE_URLS";
     private const string DashboardUrlDefaultValue = "http://localhost:18888";
 
     private readonly WebApplication _app;
+    private Func<IPEndPoint>? _browserEndPointAccessor;
+    private Func<IPEndPoint>? _otlpServiceEndPointAccessor;
 
-    public DashboardWebApplication()
+    public Func<IPEndPoint> BrowserEndPointAccessor
+    {
+        get => _browserEndPointAccessor ?? throw new InvalidOperationException("WebApplication not started yet.");
+    }
+    public Func<IPEndPoint> OtlpServiceEndPointAccessor
+    {
+        get => _otlpServiceEndPointAccessor ?? throw new InvalidOperationException("WebApplication not started yet.");
+    }
+    public DashboardWebApplication(Action<WebApplicationBuilder>? configureBuilder = null)
```

Comment:
Perhaps document that this parameter is only for testing purposes.

Also some white space around the prior two properties.